hdf5/src/H5Fint.c

```diff
@@ -299,10 +306,12 @@ H5F__set_vol_conn(H5F_t *file)
     /* Cache the connector ID & info for the container */
     file->shared->vol_id   = connector_prop.connector_id;
     file->shared->vol_info = new_connector_info;
-    if (H5I_inc_ref(file->shared->vol_id, FALSE) < 0)
-        HGOTO_ERROR(H5E_FILE, H5E_CANTINC, FAIL, "incrementing VOL connector ID failed");
 
 done:
+    if (ret_value < 0 && conn_id_incr)
+        if (H5I_dec_ref(connector_prop.connector_id) < 0)
```

Comment:
The ref count of the connector class not being decremented on failure seems like it was a pre-existing problem even in the serial case.